tinygrad/mlops.py

```diff
@@ -150,6 +150,7 @@ def backward(self, grad_output:LazyBuffer) -> Tuple[None, Optional[LazyBuffer],
 class Sum(Function):
   def forward(self, x:LazyBuffer, new_shape:Tuple[int, ...]) -> LazyBuffer:
     self.input_shape = x.shape
+    if x.dtype == dtypes.bool: x = x.cast(dtypes.int64)
```

Comment:
i think we want int32, we also differ from torch about index default (we use int32, jax use int32, torch uses int64)

Reply:
also is this output dtype tested?

Reply:
Wait, why do we want a hack like this at all?

Reply:
my bad i thought sum of int8 was int8, it can always be `least_upper_dtype(dtype, dtypes.int32)`